ctest_2_3: disable warn as error

paparodeo opened the pull request:
the development build is enabled for tests but also enables warn as
error which creates an error when building the tests on clang-19.

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

@emilazy

paparodeo commented on pkgs/development/libraries/catch2/3.nix:
```diff
@@ -31,13 +33,13 @@ stdenv.mkDerivation rec {
-    NIX_CFLAGS_COMPILE = "-Wno-error=cast-align";
-  };
+    ++ lib.optional (stdenv.hostPlatform.isRiscV || stdenv.hostPlatform.isAarch32) "-Wno-error=cast-align"
+    ++ lib.optional stdenv.cc.isClang "-Wno-error=c++20-extensions"
```
yeah, maybe but I didn't want to change the behavior from upstream. I'm not going to test it to see if using c++20 is going to break their coding assumptions.

paparodeo replied:
and the base code builds fine without it its just that we compile with `CATCH_DEVELOPMENT_BUILD=ON` for some reason which adds the code that triggers the error, IIRC, so disabling the warning seemed minimally invasive.

emilazy replied:
They support C++20, e.g.

* https://github.com/catchorg/Catch2/blob/7bbd4b90750ea20a08377c600da924d5d6d7f8ef/docs/release-notes.md?plain=1#L326-L328
* https://github.com/catchorg/Catch2/blob/7bbd4b90750ea20a08377c600da924d5d6d7f8ef/src/catch2/internal/catch_meta.hpp#L31-L33
* https://github.com/catchorg/Catch2/blob/7bbd4b90750ea20a08377c600da924d5d6d7f8ef/tests/SelfTest/UsageTests/Message.tests.cpp#L244-L245

But I guess we turn on the development build because [it enables tests](https://github.com/catchorg/Catch2/blob/7bbd4b90750ea20a08377c600da924d5d6d7f8ef/docs/cmake-integration.md). We should just toggle `CATCH_ENABLE_WERROR` back off, since that’s the default behaviour otherwise and it’s harmful rather than useful for us.

paparodeo committed:
ctest_2_3: disable warn as error

the development build is enabled for tests but also enables warn as
error which creates an error when building the tests on clang-19.
